raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -1072,6 +1073,48 @@ def test_events_for_close():
         assert pair.payer_state == 'payer_waiting_close'
 
 
+def test_events_for_onchain_secretreveal():
+    """ Secret must be registered on-chain when the unsafe region is reached and
+    the secret is known.
+    """
+
+    amount = 10
+    channelmap, transfers_pair = make_transfers_pair(
+        [HOP2_KEY, HOP3_KEY],
+        amount,
+    )
+
+    pair = transfers_pair[0]
+    channel_identifier = pair.payer_transfer.balance_proof.channel_address
+    channel_state = channelmap[channel_identifier]
+
+    # Reveal the secret off-chain
+    for channel_state in channelmap.values():
+        channel.register_secret(channel_state, UNIT_SECRET, UNIT_SECRETHASH)
+
+    block_number = (
+        pair.payer_transfer.lock.expiration - channel_state.reveal_timeout
+    )
+
+    # If we are not in the unsafe region, we must NOT emit ContractSendSecretReveal
+    events = mediator.events_for_onchain_secretreveal(
+        channelmap,
+        transfers_pair,
+        block_number - 1,
+    )
+    assert not events
+
+    # If we are in the unsafe region, we must emit ContractSendSecretReveal
+    events = mediator.events_for_onchain_secretreveal(
+        channelmap,
+        transfers_pair,
+        block_number,
+    )
+
+    assert isinstance(events[0], ContractSendSecretReveal)
```

Comment:
please use the test utility [`must_contain_entry`](https://github.com/raiden-network/raiden/blob/210c3b25381cc95735fe62991267d22f140be2dc/raiden/tests/utils/events.py#L21), instead of a fixed index, if a new event is added later to this function the tests will unnecessarily fail.

Example usage: https://github.com/raiden-network/raiden/blob/master/raiden/tests/integration/transfer/test_directtransfer_statechanges.py#L41-L45